### PR TITLE
Anvil integration

### PR DESCRIPTION
**Pull request by martinvol**
This should be merged as soon as possible.

Known issues:

- [ ] StableTokens are deployed but not all exchanges are deployed, only one exchange (Mento) is deployed
- [ ] pretify pre-commit hook breaks when comiting Solidity
- [ ] Not all precompiles implemented, need to create loop to deploy them
- [x] validators test still doesn't work
- [x] tests are failing
- [ ] script is still not split (run migrations, run with integration tests, etc)
- [ ] Initialization interfaces need to be defined
- [ ] json doesn't support full config (for example, sender keys are not configurable)
- [ ] validator deployment is oversimplied (din't think it was worth migrating the whole thing)
- [ ] Create a central list of core contracts in solidity
- [ ] Supply of eth not doesn't match what the Celo smart contract thinks it is.
- [ ] Standalone CI job to generate migrations not created.
- [ ] There's a problem with compile times. Libraries are deployed, then contracts recompiled for linking BUT forge for some reason recompiles everything, and takes time.
- [ ] Consider breaking migrations.s.sol into multiple files

### CODE REVIEW

File: packages/protocol/.env.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-  "celoScanApiKey": "[Celo scan api]"
+  "celoScanApiKey": "VR3SY2MYRB1RJVQQV4WQ8B4BT6NAHIX1KZ"
```
Comment (martinvol) [inline packages/protocol/.env.json]: This key has been rotated

File: remappings.txt [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ds-test/=packages/protocol/lib/celo-foundry/lib/forge-std/lib/ds-test/src/
 forge-std/=packages/protocol/lib/celo-foundry/lib/forge-std/src/
+forge-std-8/=packages/protocol/lib/celo-foundry-8/lib/forge-std/src/
```
Comment (martinvol) [inline remappings.txt:3]: I wonder why we had this package if it was previously unused

Comment (soloseng) [reply remappings.txt:3]: This is used in `foundry.toml`. It was added to test GasPriceMinimum

**Review by pahor167 — APPROVED** [review]

**Comment by arthurgousset** [comment]
(for visibility) @soloseng moved the todo list into a separate issue: 
- https://github.com/celo-org/celo-monorepo/issues/10990

**Pull request merged**